pkg/discovery/oci_dbbacked.go

```diff
@@ -133,8 +138,32 @@ func (od *DBBackedOCIDiscovery) listGroupsFromInventory() ([]*plugininventory.Pl
 // inventory of this discovery and stores it in the cache directory.
 func (od *DBBackedOCIDiscovery) fetchInventoryImage() error {
 	// TODO(khouzam): Improve by checking if we really need to download again or if we can use the cache
+
+	// Get the custom public key path and prepare cosign verifier, if empty, cosign verifier would use embedded public key for verification
+	customPublicKeyPath := os.Getenv(constants.CustomPublicKeyPathForPluginMetadataSignature)
+	cosignVerifier := cosignhelper.NewCosignVerifier(customPublicKeyPath)
+	if err := od.verifyInventoryImageSignature(cosignVerifier); err != nil {
+		log.Warningf("Unable to verify Plugin metadata signature: %v", err)
+		log.Warningf("Warning, Plugin metadata signature verification failed and therefore CLI couldn't ensure the integrity of the plugins to be installed. If you choose to ignore the validation please set environment variable SKIP_PLUGIN_METADATA_SIGNATURE_VERIFY to true ")
+	}
+
 	if err := carvelhelpers.DownloadImageAndSaveFilesToDir(od.image, od.pluginDataDir); err != nil {
 		return errors.Wrapf(err, "failed to download OCI image from discovery '%s'", od.Name())
 	}
 	return nil
 }
+
+func (od *DBBackedOCIDiscovery) verifyInventoryImageSignature(verifier cosignhelper.Cosignhelper) error {
+	skipPluginDBSignatureVerify := os.Getenv(constants.SkipPluginMetadataSignatureVerification)
+	if skipPluginDBSignatureVerify != "" {
+		if skip, _ := strconv.ParseBool(skipPluginDBSignatureVerify); skip {
+			log.V(9).Info("Skipping the signature verification\n")
```

Comment:
Initially I planned to show this message always, but was not convinced as the user explicitly choosen it to ignore. However I see your point, it's always better to show this message(as warning may be) so that user is reminded of the insecure settings he had chosen so that he can attempt to fix it soon. I will update it to show it always.